servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/OutlierDetectorFactory.java

```diff
@@ -24,7 +24,7 @@
  * @param <ResolvedAddress> the type of the resolved address.
  * @param <C> the type of the load balanced connection.
  */
-public interface OutlierDetectorFactory<ResolvedAddress, C extends LoadBalancedConnection> {
+interface OutlierDetectorFactory<ResolvedAddress, C extends LoadBalancedConnection> {
```

Comment:
Since we don't need to consume these with `LoadBalancerBuilder` there is no reason to make this public anymore. We can always do so later.

Reply:
`XdsOutlierDetectorFactory` is not used, can it be removed? Can we actually remove the whole `OutlierDetectorFactory`? It's used only for testing, can we move it to `test` scope?

Reply:
We can remove it for now.